lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHook.java

```diff
@@ -182,70 +183,69 @@ private NotificationEntity makeNotificationEntityWithIncludedFields(HookDoc hook
-                                                                 node.asText()));
-                }
-            }
-        }
-        // Add entity identities to properties
+        List<PathAndValue> entityData=new ArrayList<>();
+        List<PathAndValue> removedEntityData = new ArrayList<>();
+        List<String> changedPaths = new ArrayList<>();
+        List<String> removedElements = new ArrayList<>();
+
+        // Add entity identities to entity data
         for (Field identityField : metadata.getEntitySchema().getIdentityFields()) {
             Path identityPath = identityField.getFullPath();
             String pathString = identityPath.toString();
-            boolean found=false;
-            for(NotificationEntity.PathAndValue v:data) {
-                if(v.getPath().equals(pathString)) {
-                    found=true;
-                    break;
-                }
-            }
-            if(!found) {
-                String valueString = postDoc.get(identityPath).asText();            
-                data.add(new NotificationEntity.PathAndValue(pathString, valueString));
-            }
+            String valueString = postDoc.get(identityPath).asText();
+            entityData.add(new PathAndValue(pathString, valueString));
         }
-        List<String> changedPaths=new ArrayList<>();
-        List<NotificationEntity.PathAndValue> removedPaths=new ArrayList<>();
-        List<String> removedElements=new ArrayList<>();
-        for(DocComparator.Delta<JsonNode> delta:diff.getDelta()) {
-            if( (delta instanceof DocComparator.Move && arrayOrderSignificant) ||
-                !(delta instanceof DocComparator.Move) ) {
-
-                if(delta instanceof DocComparator.Removal) {
-                    JsonNode removedNode=((DocComparator.Removal<JsonNode>)delta).getRemovedNode();
-                    if(removedNode.isContainerNode()) {
-                        removedElements.add(delta.getField().toString());
-                        flatten(delta.getField().toString(),
-                                ((DocComparator.Removal<JsonNode>)delta).getRemovedNode(),
-                                removedPaths);
-                    } else {
-                        removedPaths.add(new NotificationEntity.PathAndValue(delta.getField().toString(),removedNode.asText()));
-                    }
+
+        // Add flattened include doc to entity data
+        JsonDoc includeDoc=includeProjector.project(postDoc,jsonNodeFactory);
+        flatten("", includeDoc.getRoot(), entityData);
+
+        // Add changes to entity data, removed entity data, changed paths, and removed elements
+        for(DocComparator.Delta<JsonNode> delta : diff.getDelta()) {
+            if (delta instanceof DocComparator.Move && arrayOrderSignificant) {
+                String newPath = delta.getField2().toString();
+                JsonNode movedNode = ((DocComparator.Move<JsonNode>) delta).getMovedNode();
```

Comment:
Prefer to see a line like this first in the block, so it doesn't get too far from `instanceof` that validates it.